src/Microsoft.AspNet.Html.Abstractions/HtmlContentBuilder.cs

```diff
@@ -1,62 +1,97 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
 using System.Text.Encodings.Web;
-using Microsoft.AspNet.Html.Abstractions;
 
-namespace Microsoft.Extensions.Internal
+namespace Microsoft.AspNet.Html.Abstractions
 {
     /// <summary>
-    /// Enumerable object collection which knows how to write itself.
+    /// An <see cref="IHtmlContentBuilder"/> implementation using an in memory list.
     /// </summary>
-    [DebuggerDisplay("{DebuggerToString()}")]
-    internal class BufferedHtmlContent : IHtmlContentBuilder
+    public class HtmlContentBuilder : IHtmlContentBuilder
     {
+        /// <summary>
+        /// Creates a new <see cref="HtmlContentBuilder"/>.
+        /// </summary>
+        public HtmlContentBuilder()
+            : this(new List<object>())
+        {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlContentBuilder"/> with the given initial capacity.
+        /// </summary>
+        /// <param name="capacity">The initial capacity of the backing store.</param>
+        public HtmlContentBuilder(int capacity)
+            : this(new List<object>(capacity))
+        {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlContentBuilder"/> with the given list of entries.
+        /// </summary>
+        /// <param name="entries">
+        /// The list of entries. The <see cref="HtmlContentBuilder"/> will use this list without making a copy.
+        /// </param>
+        public HtmlContentBuilder(List<object> entries)
```

Comment:
The idea here is that if you want to use this as a wrapper around pre-allocated concept. It could just as easily be `IList<T>` if that makes you happier.